app/core/src/main/java/com/fsck/k9/controller/push/PushController.kt

```diff
@@ -142,17 +150,24 @@ class PushController internal constructor(
         }
     }
 
+    @Suppress("LongMethod", "CyclomaticComplexMethod")
     private fun updatePushers() {
         Timber.v("PushController.updatePushers()")
 
         val generalSettings = generalSettingsManager.getSettings()
 
+        val alarmPermissionMissing = !alarmPermissionManager.canScheduleExactAlarms()
         val backgroundSyncDisabledViaSystem = autoSyncManager.isAutoSyncDisabled
         val backgroundSyncDisabledInApp = generalSettings.backgroundSync == BackgroundSync.NEVER
         val networkNotAvailable = !connectivityManager.isNetworkAvailable()
         val realPushAccounts = getPushAccounts()
 
-        val pushAccounts = if (backgroundSyncDisabledViaSystem || backgroundSyncDisabledInApp || networkNotAvailable) {
+        val pushAccounts = if (
+            backgroundSyncDisabledViaSystem ||
+            backgroundSyncDisabledInApp ||
+            networkNotAvailable ||
+            alarmPermissionMissing
+        ) {
```

Comment:
This could be changed for better readability:

```suggestion
        val shouldDisablePushAccounts = backgroundSyncDisabledViaSystem ||
                backgroundSyncDisabledInApp ||
                networkNotAvailable ||
                alarmPermissionMissing
        
        val pushAccounts = if (shouldDisablePushAccounts) {
```